openedx/core/storage.py

```diff
@@ -112,5 +113,14 @@ class name is not given, an instance of the default storage is returned.
     the storage implementation makes http requests when instantiated, for
     example.
     """
-    storage_cls = import_string(storage_class or settings.DEFAULT_FILE_STORAGE)
-    return storage_cls(**kwargs)
+    if storage_class is not None:
+        storage_cls = import_string(storage_class)
+        return storage_cls(**kwargs)
+
+    default_storage = storages["default"]
+    if kwargs:
+        # If custom kwargs provided, create new instance
+        storage_cls = default_storage.__class__
+        return storage_cls(**kwargs)
+
+    return default_storage
```

Comment:
Again, this is a lot of code just to the handle the upgrade of `settings.DEFAULT_FILE_STORAGE` to `settings.STORAGES['default']['BACKEND']`. What is the purpose of the new logic?